src/core/mount.h

```diff
@@ -36,9 +36,10 @@ typedef struct MountParameters {
 
 /* Used while looking for mount points that vanished or got added from/to /proc/self/mountinfo */
 typedef enum MountProcFlags {
-        MOUNT_PROC_IS_MOUNTED   = 1 << 0,
-        MOUNT_PROC_JUST_MOUNTED = 1 << 1,
-        MOUNT_PROC_JUST_CHANGED = 1 << 2,
+        MOUNT_PROC_IS_MOUNTED      = 1 << 0,
+        MOUNT_PROC_JUST_MOUNTED    = 1 << 1,
+        MOUNT_PROC_JUST_CHANGED    = 1 << 2,
+        MOUNT_PROC_DEVICE_CHANGED  = 1 << 3,
```

Comment:
comments would be good here, that clarify the difference between the two flags at the end, i.e. that the former is about any chance, the latter about just the "what" part.

actually, I think the latter flag should probably be named more precisely. i.e. the name "device" only really makes sense if we really talk about device nodes here, but mounts can be of nfs servers and whatnot. We typically call the concept that is compared here the "what" field. the mount(2) man page calls it the "source" field. 

hence I#d propose:

MOUNT_PROC_JUST_CHANGED_WHAT or MOUNT_PROC_JUST_CHANGED_SOURCE

(it's a special case of the MOUNT_PROC_JUST_CHANGED flag, after all)